awx/main/scheduler/task_manager_models.py

```diff
@@ -0,0 +1,120 @@
+# Copyright (c) 2022 Ansible by Red Hat
+# All Rights Reserved.
+import logging
+
+from django.conf import settings
+
+from awx.main.models import (
+    Instance,
+    InstanceGroup,
+)
+
+logger = logging.getLogger('awx.main.scheduler')
+
+
+class TaskManagerInstance:
+    """A class representing minimal data the task manager needs to represent an Instance."""
+
+    def __init__(self, obj, node_type, capacity, hostname):
+        self.obj = obj
+        self.node_type = node_type
+        self.remaining_capacity = capacity
+        self.capacity = capacity
+        self.hostname = hostname
+
+
+class TaskManagerInstances:
+    def __init__(self, active_tasks=[]):
```

Comment:
Another mutable default `[]`. If it's really _required_ then you could have it as a positional argument. Positional arguments aren't bad when there's only 1. Or you could allow `None`, in which case you don't fill in capacities.